### PR TITLE
Fixes for gc related tests

### LINKED ISSUE #3998: GC_WORKLOAD variable is missing in e2e conf to support gc test cases
/kind bug

**What steps did you take and what happened:**
gc testcases will apply some workloads for the testing. It will read the yaml from GC_WORKLOAD variable. We should provide the variables in e2e conf, otherwise the testcases would fail with exception.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by wyike**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3998

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: test/e2e/suites/gc_managed/gc_managed_test.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ var _ = ginkgo.Describe("[managed] [gc] EKS Cluster external resource GC tests",
 		cp := ms.GetControlPlaneByName(ctx, ms.GetControlPlaneByNameInput{
 			Getter:    e2eCtx.Environment.BootstrapClusterProxy.GetClient(),
 			Namespace: cluster.Spec.InfrastructureRef.Namespace,
-			Name:      cluster.Spec.InfrastructureRef.Name,
+			Name:      cluster.Spec.ControlPlaneRef.Name,
```
Comment (wyike) [inline test/e2e/suites/gc_managed/gc_managed_test.go:91]:
```
Control Plane Ref:
    API Version:  controlplane.cluster.x-k8s.io/v1beta2
    Kind:         AWSManagedControlPlane
    Name:         eks-extresgc-41ehz7-control-plane
    Namespace:    eks-extresgc-3lx403
  Infrastructure Ref:
    API Version:  infrastructure.cluster.x-k8s.io/v1beta2
    Kind:         AWSManagedCluster
    Name:         eks-extresgc-41ehz7
    Namespace:    eks-extresgc-3lx403
```
Get the controlplane name from ControlPlaneRef

**Comment by richardcase** [comment]
@wyike - is there much more to do on this?

**Comment by richardcase** [comment]
> it seems AUTO_CONTROLLER_IDENTITY_CREATOR=true won't affect AWSClusterStaticIdentity functionalities, right?

Thats correct @wyike 👍

**Pull request merged**